dspace-api/src/main/java/org/dspace/xmlworkflow/storedcomponents/dao/impl/XmlWorkflowItemDAOImpl.java

```diff
@@ -34,21 +38,23 @@ protected XmlWorkflowItemDAOImpl() {
     }
 
     @Override
-    public List<XmlWorkflowItem> findAllInCollection(Context context, Integer offset, Integer limit,
-                                                     Collection collection) throws SQLException {
-        Criteria criteria = createCriteria(context, XmlWorkflowItem.class);
+    public List<XmlWorkflowItem> findAllInCollection(Context context, Integer offset, Integer limit, Collection
+        collection) throws SQLException {
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, XmlWorkflowItem.class);
+        Root<XmlWorkflowItem> xmlWorkflowItemRoot = criteriaQuery.from(XmlWorkflowItem.class);
+        criteriaQuery.select(xmlWorkflowItemRoot);
         if (collection != null) {
-            criteria.add(Restrictions.eq("collection", collection));
+            criteriaQuery.where(criteriaBuilder.equal(xmlWorkflowItemRoot.get(XmlWorkflowItem_.collection),
+                collection));
         }
-
-        if (offset != null) {
-            criteria.setFirstResult(offset);
+        if (offset == null) {
+            offset = -1;
         }
         if (limit != null) {
-            criteria.setMaxResults(limit);
+            limit = -1;
```

Comment:
This looks different than the current code.  Should this be `if (limit == null) { limit = -1; }`?